Tchap/Managers/PublicRoom/PublicRoomService.swift

```diff
@@ -0,0 +1,99 @@
+/*
+ Copyright 2018 New Vector Ltd
+ 
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import Foundation
+import RxSwift
+
+/// `PublicRoomService` implementation of `PublicRoomServiceType` is used to get public rooms from Tchap platforms.
+final class PublicRoomService: PublicRoomServiceType {
+    
+    // MARK: - Properties
+    
+    private let session: MXSession
+    private let homeServersStringURL: [String]
+    
+    private var currentHomeServer: String? {
+        return self.session.matrixRestClient.homeserver
+    }
+    
+    // MARK: - Setup
+    
+    init(homeServersStringURL: [String], session: MXSession) {
+        self.homeServersStringURL = homeServersStringURL
```

Comment:
I would suggest:
1- extract the user homeserver name by removing the known prefix "https://matrix.".
2- remove it from the provided list
3- add an empty string to the list (this empty string should be replaced by nil during the map)

Note: the current user HS may not be present in the provided list. We are supposed to add it in this case.